taf/repository_tool.py

```diff
@@ -447,14 +411,22 @@ def all_target_files(self):
                     )
         return targets
 
-    def _check_if_files_delegated_to_role(self, targets_role, target_files):
-        target_roles_mapping = self.map_signing_roles(target_files)
-        roles = set(target_roles_mapping.values())
-        if len(roles) > 1 or targets_role not in roles:
-            raise TargetsError(
-                f"Target files {', '.join(target_files)} delegated to {', '.join(roles)} "
-                f"and not just {targets_role}"
-            )
+    def get_role_from_target_paths(self, target_paths):
+        """
+        Find a common role that could be used to sign given target paths.
+        """
+        targets_roles = self.map_signing_roles(target_paths)
+        roles = list(targets_roles.values())
+
+        # all target files should have at least one common role
```

Comment:
Yup, I knew that we have a single role for a target path, just wanted to implement this in case for future needs (since tuf supports multiple roles per target path). I'll add comments and tests.